### PR TITLE
nanocoap_cache: port to ztimer

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
In the latest release 2022.01, a lot of effort was put into porting most modules to ztimer, so let's just use it with the CoAP cache as well :-).

### CODE REVIEW

**Comment by miri64** [comment]
The last commit is an extra. Not sure if the compiler rules in RIOT 2022.01 were made stricter or if my GCC just became stricter via an update, but with the last commit `nanocoap_cache` becomes buildable in 2022.01 again.

**Review by cgundogan — APPROVED** [review]
Looks good!  Thanks for the adaptations + fixes.

File: sys/net/application_layer/nanocoap/cache.c [hunk 1]
```diff
@@ -202,15 +211,15 @@ int nanocoap_cache_process(const uint8_t *cache_key, unsigned request_method,
...
     }
     /* This response is cacheable: Caches can use the Max-Age Option
        to determine freshness and (if present) the
        ETag Option for validation.
     */
     else if (resp->hdr->code == COAP_CODE_CONTENT) {
-        uint64_t now = xtimer_now_usec64();
+        uint32_t now = ztimer_now(ZTIMER_SEC);
         /* cache entry is stale */
         if (ce && (ce->max_age < now)) {
```
Comment (kaspar030) [inline sys/net/application_layer/nanocoap/cache.c:224]: this maybe won't be "considered stale immediately" with ZTIMER_SEC?

Comment (miri64) [reply sys/net/application_layer/nanocoap/cache.c:224]: wrt to https://github.com/RIOT-OS/RIOT/issues/17808: this line was removed in https://github.com/RIOT-OS/RIOT/pull/17801/commits/6b3a2729f9910f3450daa87c6c1ca0170e29b6b8 (which is soon to be cherry-picked into https://github.com/RIOT-OS/RIOT/pull/13889), and was thus not passed in the setup described in https://github.com/RIOT-OS/RIOT/issues/17808.

**Pull request merged**